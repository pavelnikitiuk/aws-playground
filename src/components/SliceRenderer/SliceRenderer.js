import { Box } from '@chakra-ui/react';
import React from 'react';

function SliceRenderer({ slices, sliceMap }) {
  return slices
    .filter(({ slice_type: sliceType }) => Boolean(sliceMap[sliceType]))
    .map(({ slice_type: sliceType, ...otherProps }) => {
      const sliceConfig = sliceMap[sliceType];
      const { Component, model } = sliceConfig;

      return (
        <Box key={otherProps.id} as="section" mb="10">
          <Component {...model(otherProps)} />
        </Box>
      );
    });
}

export default SliceRenderer;
