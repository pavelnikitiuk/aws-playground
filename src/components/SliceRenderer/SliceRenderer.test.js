import React from 'react';
import renderer from 'react-test-renderer';
import SliceRenderer from './index';

const MOCK_COMPONENT = jest.fn().mockReturnValue('it is test a');
const MOCK_MODEL = jest.fn();

const MOCK_SLICE_DATA = [
  {
    id: 1,
    slice_type: 'test-1',
  },
  {
    id: 2,
    slice_type: 'test-2',
  },
];

const MOCK_SLICE_MAP = {
  'test-1': {
    Component: MOCK_COMPONENT,
    model: MOCK_MODEL,
  },
};

describe('Given the SliceRenderer component', () => {
  describe('when it is rendered', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = renderer.create(
        <SliceRenderer slices={MOCK_SLICE_DATA} sliceMap={MOCK_SLICE_MAP} />,
      );
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should create model from passed data', () => {
      expect(MOCK_MODEL).toHaveBeenCalledWith({ id: 1 });
    });
  });
});
