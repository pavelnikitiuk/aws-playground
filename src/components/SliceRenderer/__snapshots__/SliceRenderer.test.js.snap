// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the SliceRenderer component when it is rendered should match the snapshot 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1r2f04i{margin-bottom:10px;}",
      }
    }
    data-emotion="css 1r2f04i"
  />,
  <section
    className="css-1r2f04i"
  >
    it is test a
  </section>,
]
`;
